airflow/dag_processing/manager.py

```diff
@@ -383,14 +383,15 @@ def __init__(
         self._dag_ids = dag_ids
         self._async_mode = async_mode
         self._parsing_start_time: Optional[int] = None
+        self._standalone_mode = conf.getboolean("scheduler", "standalone_dag_processor")
 
         # Set the signal conn in to non-blocking mode, so that attempting to
         # send when the buffer is full errors, rather than hangs for-ever
         # attempting to send (this is to avoid deadlocks!)
         #
         # Don't do this in sync_mode, as we _need_ the DagParsingStat sent to
         # continue the scheduler
-        if self._async_mode:
+        if self._async_mode and not self._standalone_mode and self._signal_conn is not None:
```

Comment:
Do we ever get the situation that we have `self._signal_conn` and 'self._standalone_mode`  both set ? I do not think so - after this change we can simplify that check to just `and not self._standalone_mode`. 

I believe after this change we only have two cases:

a) Standalone mode -> no  _signal_conn set
b) Non-standalone mode -> _signal_conn  set

Reply:
Makes sense, Removed _standalone_mode and rely only on _signal_conn (actually renamed to _direct_scheduler_conn)